ldk/javascript/src/types/oliveHelps/index.d.ts

```diff
@@ -300,56 +283,56 @@ declare namespace OliveHelps {
   type WhisperHandler = (error: Error | undefined, whisper: Whisper) => void;
   type WhisperHandlerWithParam<T> = (error: Error | undefined, param: T, whisper: Whisper) => void;
 
-  type Button = Component<WhisperComponentType.Button> & {
+  type Button = Component<'button'> & {
```

Comment:
Why can it not be "WhisperComponentType.button"? I thought having string defined in one place and using strongly typed param would reduce potential future issues.

Reply:
The Loop author continues to use `whisper.WhisperComponentType.Button` I do believe.

Reply:
OK so this is just the change on the internal representation, so it doesn't affect the Loop author's experience like Cory mentioned.

It can't be `WhisperComponentType.Button` because that enum doesn't exist (never did). What I could've done was something like this:
```typescript
type WhisperTypePropMarkdown = 'markdown';
type WhisperTypePropButton = 'button';
...
type WhisperComponentType = 
  | WhisperTypePropMarkdown
  | WhisperTypePropButton;
```

But when I started playing around with it it felt pretty awkward to use and type, so I ended up defining `OliveHelps.WhisperComponentType` as a union type of all the possible strings:
```typescript
type WhisperComponentType = 
  | 'markdown'
  | 'button';
```

So while the param is still strongly typed, the difference is that a) it doesn't refer to something that doesn't exist, and b) the only way this becomes a problem is if we repurpose one of the existing strings for something else entirely, which we won't do.

Reply:
I understand your point. I think what took my attention was that we are using same strings in 2 different places like ('button')
`type Button = Component<'button'> & {` 
and 
`type WhisperComponentType =
    | 'box'
    | 'button'`

Wondering if it could lead to an issue if we decide to update a string to something else. Preferably it would be defined in 1 place and we could have them strongly typed. But this is minor and I don't think I'm strongly stand on it ,so I will leave it up to you. Just want to make sure I explain better what I meant.

Reply:
I think I basically agree with everything you said above in analyzing the problem and potential pitfalls. It could go either way, and it's a judgment call and if the team feels strongly about the other way I'll go with that too. I think my logic here is that these string values are very unlikely to be mutated. We will never rename a `'button'` to `'specialButton'`, as an example. Adding a separate type declaration adds a bit of indirection, so you need to unpeel that to get the underlying value.